Makefile

```diff
@@ -12,8 +12,10 @@ BINARY=keymaster
 # These are the values we want to pass for Version and BuildTime
 VERSION=1.15.3
 DEFAULT_HOST?=
-DEFAULT_LDFLAGS=-X main.Version=${VERSION}
-CLIENT_LDFLAGS=${DEFAULT_LDFLAGS} -X main.defaultHost=${DEFAULT_HOST}
+VERSION_FLAVOUR?=cf
```

Comment:
I don't think the "upstream" version should have a flavour. It's the default. This should be the empty string.

Reply:
can it be main / upstream? Otherwise it prints "1.15.3-" as the version

Reply:
This can be solved by having the flavour string that's injected contain a leading `-`.

Reply:
ok changed

Reply:
Why so many changes? I would have done this with a new conditional on the makefile and a new variable:

PRINTVERSION=version
ifeq ($(FLAVOR),)
   PRINTVERTION = ${VERSION}-${FLAVOR}

And then use the printversion to the build instead of version.


Also I dont like using the useragent string for something else.

Reply:
Thats doable as well :) so maybe lets close this PR and when you have it done I will merge it to the PS fork.

Reply:
Or modify this PR?

Reply:
modify this PR :)

Reply:
done @cviecco

Reply:
Also, I still believe that useragent is better for debugging than the plain version as it contains also OS and Arch. If somebody pastes this in the report, it saves time / roundtrips for questions which OS do you use.